lte/gateway/python/integ_tests/s1aptests/util/traffic_util.py

```diff
@@ -566,37 +600,68 @@ def combine(test, *tests):
 
     @property
     def results(self):
+        """Return the traffic results data"""
         return self._results
 
     def start(self):
-        ''' Start this test by spinning off runner thread '''
+        """Start this test by spinning off runner thread"""
         self._done.clear()
         threading.Thread(target=self._run).start()
 
     def verify(self):
-        ''' Verify the results of this test
+        """Verify the results of this test
 
-        Raises a RuntimeError if any tests returned with an error message
-        '''
+        Raises:
+            RuntimeError: if any tests returned with an error message
+        """
         self.wait()
         with self._test_lock:
             if not isinstance(self.results, tuple):
+                if not self._done.is_set():
+                    TrafficUtil.need_to_close_iperf3_server = True
+                    self._done.set()
+                    self.cleanup()
                 raise RuntimeError(
-                    'Cached results object is not a tuple : {}'.format(
+                    "Cached results object is not a tuple : {0}".format(
                         self.results,
                     ),
                 )
             for result in self.results:
                 if not isinstance(result, iperf3.TestResult):
                     raise RuntimeError(
-                        'Cached results are not iperf3.TestResult objects',
+                        "Cached results are not iperf3.TestResult objects",
                     )
                 if result.error:
+                    TrafficUtil.need_to_close_iperf3_server = True
                     # iPerf dumps out-of-order packet information on stderr,
                     # ignore these while verifying the test results
                     if "OUT OF ORDER" not in result.error:
                         raise RuntimeError(result.error)
 
     def wait(self):
-        ''' Wait for this test to complete '''
+        """Wait for this test to complete"""
         self._done.wait(timeout=TRAFFIC_TEST_TIMEOUT_SEC)
+
+    def cleanup(self):
+        """Cleanup sockets and network interfaces"""
+        # Signal to end connection
+        msg = TrafficRequest(TrafficRequestType.EXIT)
+        msg.send(self.sc_out)
+
+        # Close out network ifaces
+        net_iface_index = TrafficTest._iproute.link_lookup(
```

Comment:
are we expecting this list to never be empty? if not, this will throw an Exception

Reply:
I have always seen the interface list non empty during my debugging. However as per the comment, I have added check for non-null list for being on safer side.